python/servo/package_commands.py

```diff
@@ -418,6 +419,7 @@ def install(
                 return 1
 
         if self.is_android():
+            assert isinstance(self.target, AndroidTarget)
```

Comment:
It would be better if we didn't have an explicit assertion here as we just checked `self.is_android()` is true. Ideally, we should be able to use `self.is_android()` as a [type gaurd](https://mypy.readthedocs.io/en/stable/type_narrowing.html#user-defined-type-guards), but we can't narrow the type of a field in a class - only a normal parameter in a function or method can be narrowed.

I would suggest we just remove bot the methods `is_android` and `is_openharmony` from `CommandBase` and instead add them as free standing functions in `build_target.py` e.g
```
def is_android(target: BuildTarget) -> TypeGuard[AndroidTarget]:
    return isinstance(self.target, AndroidTarget)
```
and use it here as `if is_android(self.target)`. This will allow us to get rid of the assertion.